pkg/plugin/restore_pvc_action_plugin.go

```diff
@@ -176,9 +177,10 @@ func (p *NewPVCRestoreItemAction) Execute(input *velero.RestoreItemActionExecute
 	bSkipPVC, err := pluginItem.SkipPVCCreation(ctx, restConfig, &pvc, p.Log)
 	if err != nil {
 		return nil, errors.WithStack(err)
-	} else if bSkipPVC {
+	} else if bSkipPVC && input.Restore.Spec.NamespaceMapping == nil {
```

Comment:
Does this assume that the restore target namespace mapping does not always exist? What happens when the target ns exists, and the target PVC also exists?

From my understanding of the issue, we have a buggy `SkipPVCCreation`, in the sense that it should be checking if target ns exists, and if so target PVC within it exists. The presence of namespacemapping in the restore spec alone does not mean that it does not exist.

There is a possibility that I am wrong, if velero explicitly states that when specifying target restore namespace mapping, the target namespace should definitely not exist, in that case your fix should be sufficient.

Reply:
If `Restore.Spec.NamespaceMapping` is nil, that means that the `--namespace-mapping` flag was not used when creating a Restore with Velero. If the target ns exists, that means the `--namespace-mapping` flag was used when creating a Restore with Velero, therefore it will not skip PVC creation whether or not the target PVC that was backed up exists in the cluster. Checking if this spec is nil in the Restore CRD ensures PVC creation in the target namespace. 

The `--namespace-mapping` flag requires a 1-to-1 mapping of the namespace being backed up and the namespace to restore the backed up resources into (e.g. `velero restore create restore-to-new-namespace --from-backup old-namespace-backup --namespace-mappings old-ns:new-ns` where `old-ns` is the backed up namespace and `new-ns` is the target namespace)

See also: https://velero.io/docs/v1.13/restore-reference/#persistent-volume-rename
>When restoring PVs, if the PV being restored does not exist on the target cluster, Velero will create the PV using the name from the backup. Velero will rename a PV before restoring if both of the following conditions are met:

>1. The PV already exists on the target cluster.
>2. The PV’s claim namespace has been [remapped](https://velero.io/docs/v1.13/restore-reference/#restoring-into-a-different-namespace).

Reply:
Updated `SkipPVCCreation` to take `targetNamespace` as an argument.